Reworks Duffel Bags (Zippers)

LemonInTheDark opened the pull request:
## About The Pull Request

Reworks duffel bags in line with oranges proposed plan.

![image](https://github.com/tgstation/tgstation/assets/58055496/126743dd-d7b8-47e0-bdd8-a0caec39c515)

Basically, instead of just making you slower all the time, they make you slower while you have them open, but give you the same speed while they're closed.
As a trade off, opening and closing them takes time, 2.1 seconds (matches the sound) and 0.5 respectively.

https://github.com/tgstation/tgstation/assets/58055496/555d2cd0-038e-4b0b-a693-0c66dac16f5b

[Adds support for limiting extra storage, uses it to make syndie stuff cool](https://github.com/tgstation/tgstation/pull/76313/commits/d0b2bbf937435b36de3ba497c48771f563b76684)
[d0b2bbf](https://github.com/tgstation/tgstation/pull/76313/commits/d0b2bbf937435b36de3ba497c48771f563b76684)

Syndicate bags currently ignore downsides by just ignoring the slowdown, but that's kinda boring so let's just buff em instead.

They now support holding a limited amount of bulky items (3), filtered down to things that would otherwise constitute going loud (or otherwise be useful to carry around as a loudish traitor)

I may have gone a bit overboard on what I whitelisted here, lemme know yeah?

I also did some fenangling with backpack uses of create_storage, I don't like this pattern it was a bad idea I think.

## Why It's Good For The Game

I'm unsure if these delays enough, I think any length of time is decent since it means you need to stop moving and focus on it for a bit.
My hope is this will make them a proper sidegrade, rather then something that goes unused/acts as newbie bait

## Changelog
:cl:
balance: Duffelbags will now only make you slow while they are unzipped. As a tradeoff, you now need to stand still and zip/unzip them to access their contents/not move real slow.
/:cl:

LemonInTheDark committed:
Reworks Duffel Bags (Zippers)

Reworks duffel bags in line with oranges proposed plan

Basically, instead of just making you slower all the time, they make you
slower while you have them open, but give you the same speed while
they're closed.
As a trade off, opening and closing them takes time, 2.1 seconds
(matches the sound) and 0.5 respectively.

I'm unsure if these are enough, I think any length of time is decent
since it means you need to stop moving and focus on it for a bit.

My hope is this will make them a proper sidegrade, rather then something
that goes unused/acts as newbie bait

LemonInTheDark committed:
Fixes desc retention, patches an issue in a unit test

LemonInTheDark committed:
fine fine

MrMelbert commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -398,11 +398,88 @@
+
+/obj/item/storage/backpack/duffelbag/update_desc(updates)
+	. = ..()
+	desc = "[initial(desc)]<br>[zipped_up ? "It's zipped up, can't get in it!" : "It's unzipped, and harder to move in."]"
```
Zipped up description sounds a bit weird, like it's intended to be a balloon alert rather than a description
```suggestion
	desc = "[initial(desc)]<br>[zipped_up ? "It's zipped up, preventing you from accessing its contents." : "It's unzipped, and harder to move in."]"
```

MrMelbert commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -398,11 +398,88 @@
+	if(!zipped_up)
+		return ..()
+
+	balloon_alert(user, "znzipping...")
```
```suggestion
	balloon_alert(user, "unzipping...")
```

MrMelbert commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -398,11 +398,88 @@
+	var/datum/callback/can_unzip = CALLBACK(src, PROC_REF(zipper_matches), TRUE)
+	if(!do_after(user, 2.1 SECONDS, src, extra_checks = can_unzip))
+		return
+	balloon_alert(user, "znzipped!")
```
```suggestion
	balloon_alert(user, "unzipped")
```

MrMelbert commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -398,11 +398,88 @@
+	var/datum/callback/can_zip = CALLBACK(src, PROC_REF(zipper_matches), FALSE)
+	if(!do_after(user, 0.5 SECONDS, src, extra_checks = can_zip))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	balloon_alert(user, "zipped!")
```
```suggestion
	balloon_alert(user, "zipped")
```
Exclamation reserved for failures

MrMelbert reviewed (dismissed):
1. Not making this a component is cowardice
2. Syndie Duffels will need attention in some way. Currently they have no innate slowdown, making them "better duffels", which this PR makes redundant / undoes.

MrMelbert commented on code/datums/actions/items/unzip.dm:
```diff
@@ -0,0 +1,16 @@
+/datum/action/item_action/zipper
+	name = "Un-Zip Duffel"
+	desc = "Un-Zip the equipped duffelbag so you can root around in it"
```
```suggestion
	name = "Unzip Duffel"
	desc = "Unzip your equipped duffelbag so you can access its contents."
```

MrMelbert commented on code/datums/actions/items/unzip.dm:
```diff
@@ -0,0 +1,16 @@
+
+/datum/action/item_action/zipper/proc/on_zip_change(datum/source, new_zip)
+	SIGNAL_HANDLER
+	var/zip = new_zip ? "Un-Zip" : "Zip"
+	name = "[zip] Duffel"
+	desc = "[zip] the equipped duffelbag so you can root around in it"
```
```suggestion
	if(new_zip)
		name = "Unzip" 
		desc = "Unzip your equipped duffelbag so you can access its contents."
	else
		name = "Zip"
		desc = "Zip your equipped duffelbag so you can move around faster."
```
Desc probably shouldn't read out "so you can root around in it" for when the action is "zipping"

LemonInTheDark replied:
ah fuck forgot about that

LemonInTheDark commented:
> 1. Not making this a component is cowardice

Component users when something is unique (I'll think about it)

>     2. Syndie Duffels will need attention in some way. Currently they have no innate slowdown, making them "better duffels", which this PR makes redundant / undoes.

Hm. I don't think I like the idea of just leaving them without any of the new stuff, vibes fucking raw.
Would need something to make them better in a unique way. Preferably something good for going loud. Will need to think.

LemonInTheDark committed:
Better button desc

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

LemonInTheDark committed:
Melbert's review, better descriptions cleaner balloon_alerts

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Likteer commented:
@LemonInTheDark How about making the syndicate duffelbags fit 1 bulky item (not taking up the entire storage)?
This might have been a terrible idea when someone suggested it for all duffelbags, but now...

With the changes to other bags, people could now reasonably assume, anyone with a syndicate duffelbag is using it to carry something bulky. 
Even if a maxcap, or a gun aren't immediately visible, the bag will become suspicious again.

SomeoneYouProbablyKnow commented:
Now to remove all non-duffel bags 😈

YuliceTheDruid commented:
I mean, I imagine the syndicate duffel would still be good because it has no slowdown even while unzipped, which would essentially make it just a Better Backpack. Another thing that I've been told though I don't know if it's true or not, is that taking items/putting items into a Syndicate Duffelbag doesn't give any notification a la putting stuff in/out of Pockets.

ATH1909 commented:
I'd personally recommend making it so that the unzipping do_after() isn't stopped by movement, so that you can actually access your duffel contents in anything vaguely resembling an emergency. You'll still have that 1.8 second delay before you can pull something out and that slowdown once you've pulled it out, but you won't be relentlessly bullied by space wind or someone walking into you.

Also, if you're making these significant mechanical changes, you NEED to remove duffel bags from the random bag option. This mechanic seems VERY confusing for someone who's, say, playing the game for the first time.

SomeoneYouProbablyKnow commented:
> Also, if you're making these significant mechanical changes, you NEED to remove duffel bags from the random bag option. This mechanic seems VERY confusing for someone who's, say, playing the game for the first time.

Gosh quite true, duffel bags already mess with new players as is. Could even remove em from the srart up menu, every vendor in every department has a duffel if you wanna take one.

Also i gotta ask you and you specifically, do you ever run out of space in regular bags? If anyone's carrying a utility ~~belt~~ bag of random stuff it's gotta be you. Because i don't think i've ever run out of bug space (as regular crew).

YuliceTheDruid commented:
> Also i gotta ask you and you specifically, do you ever run out of space in regular bags? If anyone's carrying a utility ~belt~ bag of random stuff it's gotta be you. Because i don't think i've ever run out of bug space (as regular crew).

If I'm ever exploring space, yeah. Otherwise, depends on if I have any luck finding Smuggler's. I know exactly what I do and don't need to carry generally, and what I can make do without, and how to condense those things as well as I can to fit in a bag. The strongest item in the game for how easy it is to obtain is a sheet of cardboard, because it makes an entire set of tools take up a normal slot in a bag. But I'm also acoustic and enjoy sorting inventory immensely so having a nicely condensed and sorted bag in and of itself is rewarding to me.

SinfulBliss commented:
Still noobie bait IMO since using items from bag quickly is way too useful. Arguably worse for noobies since it's yet another mechanic they'd need to learn how to use to play game. I think the slowdown is more intuitive to be honest

SomeoneYouProbablyKnow commented:
> But I'm also acoustic and enjoy sorting inventory immensely so having a nicely condensed and sorted bag in and of itself is rewarding to me.

I do love a well organized bag but i usually top off at like 2 containers and a belt. But yeah with boxes as they are there is absolutely no reason to go out of your way to get extra inventory. 
Maybe if we killed the fuck out of boxes, but that's probabaly not happening 

Side-note : Observing people's inventories is terrifying, it's either nothing or 4 half filled boxes with no rhyme or reason. People really do be living like that



> Still noobie bait IMO since using items from bag quickly is way too useful. Arguably worse for noobies since it's yet another mechanic they'd need to learn how to use to play game. I think the slowdown is more intuitive to be honest

You COULD argue that this is at least easier to notice than the constant slowdown, lots of people that carry duffels got no clue that it's any different from a bag. But really duffels belong in vendors, not loudup menus

optimumtact commented:
yes let the syndicate duffel carry a gun

LemonInTheDark commented:
> @LemonInTheDark How about making the syndicate duffelbags fit 1 bulky item (not taking up the entire storage)? This might have been a terrible idea when someone suggested it for all duffelbags, but now...
> 
> With the changes to other bags, people could now reasonably assume, anyone with a syndicate duffelbag is using it to carry something bulky. Even if a maxcap, or a gun aren't immediately visible, the bag will become suspicious again.

100% on board thank you

LemonInTheDark commented:
> I'd personally recommend making it so that the unzipping do_after() isn't stopped by movement, so that you can actually access your duffel contents in anything vaguely resembling an emergency. You'll still have that 1.8 second delay before you can pull something out and that slowdown once you've pulled it out, but you won't be relentlessly bullied by space wind or someone walking into you.

I don't think I want this. Might be too punishing, but I'd like working with your inventory to be something you need to commit to, rather then something you can do easily. A slight delay on opening would just mean you run, open, suffer maybe a second of slowdown while moving, and then close it again.

I want this to be something that feels meaningful and impacts how you play, while not being a strict downgrade.

> Also, if you're making these significant mechanical changes, you NEED to remove duffel bags from the random bag option. This mechanic seems VERY confusing for someone who's, say, playing the game for the first time.

I don't think it will be that bad. You're forced to click the bag to open it, which makes the hard part more clear, and there's a visible action button in the top left to make closing a clear option.
Might need some bespoke sprites for those, but I think it'll be alright. Would be reasonably trivial to do if it's a problem but I'd like to give it a chance.

LemonInTheDark committed:
Adds support for limiting extra storage, uses it to make syndie stuff cool

Syndicate bags currently ignore downsides by just ignoring the slowdown,
but that's kinda boring so let's just buff em instead.

They now support holding a limited amount of bulky items (3), filtered down
to things that would otherwise constitute going loud (or otherwise be
useful to carry around as a loudish traitor)

I may have gone a bit overboard on what I whitelisted here, lemme know
yeah?

LemonInTheDark committed:
Merge branch 'unduffable' of https://github.com/LemonInTheDark/tgstation into unduffable

LemonInTheDark commented:
Syndi now support holding a limited amount of bulky items (3), filtered down to things that would otherwise constitute going loud (or otherwise be useful to carry around as a loudish traitor)

I may have gone a bit overboard on what I whitelisted here, lemme know yeah?

AlamoTurtle commented:
I feel it depends on the amount of actual space it takes up, so if it can only JUST squeeze in three bulkies without space for any other items, it's clear that you're banking your entire plan on those bulky items, which encourages more planning. Some badass with a duffel bag screams they prepared just what they need to do the job, the right tools and all that. Ofc I like the idea of it being a round-start selectable in the player menu (the loadout system is said to be planned, so personally I think the idea of it being tested with what little loadout system we do have is a neat idea). All in all, I look forward to this a LOT, and it gives the feel that you pack up everything you need for the job you're doing and unzip when you're ready to get to work.

I also really like the idea of people considering wearing fannypacks and such to have instant-access weaponry to pair with the duffel bag, for anyone who might not be a syndicate agent with the luxury of chameleon belts.

ATH1909 commented:
> > I'd personally recommend making it so that the unzipping do_after() isn't stopped by movement, so that you can actually access your duffel contents in anything vaguely resembling an emergency. You'll still have that 1.8 second delay before you can pull something out and that slowdown once you've pulled it out, but you won't be relentlessly bullied by space wind or someone walking into you.
> 
> I don't think I want this. Might be too punishing, but I'd like working with your inventory to be something you need to commit to, rather then something you can do easily. A slight delay on opening would just mean you run, open, suffer maybe a second of slowdown while moving, and then close it again.

the penalty isn't just the slowdown here, though- it's also a 1.8 second delay in the withdrawal of that item (bc you need to unzip the duffel to access it), which could be critical in the case of a weapon or epipen

LemonInTheDark commented:
@MrMelbert I don't think I can componentize this effectively, because the hooks I would use happen concurrently with storage stuff.
